Introduced new Slider component.

shaavan opened the pull request:
Fixes #192

- This will allow displaying a horizontal slider when the window size is smaller than 767px. And otherwise, show the children in 5 column grid style.

- Used the Slider component in Signup.tsx file.

**Screenshot:**

_Introduced dummy avatars to show the difference._

|Desktop|Mobile|
|---------|------|
|<img width="555" alt="Screenshot 2023-02-27 at 8 06 42 PM" src="https://user-images.githubusercontent.com/85434418/221595748-26f91347-3ca1-4033-af02-3b63770d3a1c.png">|<img width="490" alt="Screenshot 2023-02-27 at 8 06 42 PM" src="https://user-images.githubusercontent.com/85434418/221596013-8558196d-3a80-4310-93cc-9d0f349a3f97.jpeg">|



## Pull Request checklist

Please review the below PR requirements:

- [x] Build was run locally, and any changes were pushed
- [x] Ensure that the title clearly describes the changes
- [x] Issue to be referenced in the PR description rather than in the commits or PR title
- [x] Clean commit history

afterburn commented:
<img width="359" alt="image" src="https://user-images.githubusercontent.com/10757768/221598328-cb315794-9ba8-431d-9ffd-8b8620d3c17a.png">

@GBKS Do you think the padding of the parent element cutting off the avatars here is acceptable or would you rather we implement a solution that makes sure the avatars only get cut off by the actual screen borders?

afterburn commented on components/Modals/SignUp.tsx:
```diff
@@ -58,21 +59,20 @@ export default function SignUpModal({ open, onClose, onConfirm }) {
         <h2 className="mb-4 text-xl font-bold">Choose an avatar</h2>
 
-        <div className="mb-5 flex cursor-pointer flex-wrap justify-between">
+        <Slider>
```
Let's rename this to ` HorizontalScrollView` or something akin to that. I'd reserve Slider for [these](https://www.w3schools.com/howto/howto_js_rangeslider.asp).

shaavan replied:
Sure!

Addressed in 339d190d6fe8f5a054e46f8e988d834c63f40c4e

afterburn commented on components/Slider.tsx:
```diff
@@ -0,0 +1,13 @@
+  return (
+    <>
+      <div
+        className="scroll mb-2 flex cursor-pointer
```
Nice, I wasn't sure if this would be possible without JS. Have you tested this on a variety of devices?

shaavan replied:
Thanks!

I have a Mac M1 pro (MacOS 13.1) and iPhone 13 (iOS 16.2). I tested this PR on both devices, and it runs perfectly.

However, we might encounter a problem with the scroll bar on the slider with Android.

On iOS (Safari and Chrome), the scrollbar automatically hides when the user is not scrolling. But I am unsure if this behavior will persist in Android and other OSes.

afterburn replied:
Seems to work fine on Android, scrollbar is visible and hides itself after a timeout:

![image](https://user-images.githubusercontent.com/10757768/221884035-edeceaaa-4218-4736-8885-2b372e558ac9.png)

Let's get this merged and fix any issues in a separate PR if they are reported.

afterburn reviewed (changes requested):
Nice work, I left one small comment

shaavan commented:
Updated from 9716d5f17b121c9ceff7080f1ac160cd5c97e3ee to 339d190d6fe8f5a054e46f8e988d834c63f40c4e ([pr204.01](https://github.com/shaavan/saving-satoshi/commits/pr204.01) -> [pr204.02](https://github.com/shaavan/saving-satoshi/commits/pr204.02), [diff](https://github.com/shaavan/saving-satoshi/compare/pr204.01..pr204.02))
Addressed @afterburn comment

Changes:

1. Rename `Slider` -> `HorizontalScrollView`.

afterburn commented:
@ecurrencyhodler feel free to merge this after your tACK 😁
